src/repart/repart.c

```diff
@@ -2691,8 +2691,11 @@ static int context_copy_from_one(Context *context, const char *src) {
                 return r;
 
         r = fd_verify_regular(fd);
-        if (r < 0)
-                return log_error_errno(r, "%s is not a file: %m", src);
+        if (r < 0) {
+                r = is_device_node(src);
```

Comment:
if we have an fd to the inode, use the fd. not the path anymore.

we generally prefer race-free behaviour: pin the inode via an fd once, then only use fd to operate on it.

also, fd_verify_regular() is just a wrapper around fstat(), and is_device_node() is just a wrapper around stat(). if the strict checking of fd_verify_regular() is too strict for this case, then maybe just implement the fstat() call directly, and do S_ISREG() || S_ISBLK() explicitly

Reply:
(also is_device_node() allows char devices. i don't think we should allow that here unless ther's a clear usecase)